### DISCUSSION TITLE
Capture multiple IR commands

### DISCUSSION THREAD

**Discussion by edevil** [bengtmartensson/IrScrutinizer#452] [Decoding issues]
I'm using an Arduino nano as sender/receiver (Girs lite 1.0.3) and IrScrutinizer 2.3.1, and can capture most commands and transmit them to devices successfully. But there's one device that I just can't seem to be able to capture/transmit the correct command.

The device is a Vodafone STB (SAGEMCOM DI362-UHD) and I've tried both the original remote and a Logitech Harmony universal remote, which both can control the device. The Harmony does not support bluetooth so I'm sure the issue is not related to that. What I think is the issue is that the remote controls send more than one IR command (there's an IR icon that blinks several times in under 1 second) and IrScrutinizer only captures the first command:
<img width="1112" alt="Screenshot 2022-02-22 at 15 37 17" src="https://user-images.githubusercontent.com/12426/155171387-b666e40c-6b70-4234-94bb-ddd3f5f39092.png">

Is there a way to capture the whole Ir sequence in this case, so that we can completely emulate the remote control?

Thank you for this program, it has already helped me a lot.

**Comment by bengtmartensson** [comment]
What happens is that the signal is not recognized as a known protocol. This can have several reasons.

To investigate, please do the following: Be sure to have selected "Invoke repeat finder" and "Invoke cleaner" under Options. Go to the Scrutinize Remote/Raw remote pane and capture all (or at least a large number of) the keys on the remote. Then do a Girr export with "Raw" included in the export. Post the Girr file as a Gist here. I will then have a close look at it.

You may also be able to use the raw signals as-is.

**Reply by edevil** [reply]
Thanks for the help, here are most of the button presses. https://gist.github.com/edevil/ad3cd44f835ab7c96db4fbc1acc6f3a2

I'll try to use the raw signals tomorrow.

**Reply by edevil** [reply]
Hey @bengtmartensson ! I've tried the raw remote capture & transmit, and again, it works for all the other devices but not this box. Is there some other debug information I can capture?

Thanks.

**Reply by bengtmartensson** [reply]
Try [this](https://gist.github.com/bengtmartensson/560f27341f142c23315628e0978a04e3).

**Reply by edevil** [reply]
I imported both as raw remote and parametric remote, but in both cases commands were not acknowledged by the box... I read somewhere that the box needs an initial command to wake up and "listen" for the actual command, hence my suspicion that multiple commands are being sent in sequence.

**Reply by bengtmartensson** [reply]
Hmm, what is "multiple commands"? Your captures are quite good, and shows a repeat (blue one in the screen dump) and an ending (green).  The only thing that _may_ be suspect is the distance between the signals; the AGirs software can max represent 65535 milli seconds. I tweaked the tolerances a bit, and then it fit with the protocol called Nokia32_endframe. This covers both the repeat and the ending. Does this answer the question on "multiple commands", or do you mean anything else? What do you mean by "IrScrutinizer only captures the first command"?

**Reply by edevil** [reply]
Disclaimer: I don't know much about IR signaling. :)

There have been many posts of users on a local VF forum unable to make universal remotes work with this box. It was mentioned that Vodafone New Zealand worked with Logitech to make Harmony remotes work (and they do), and it required something to the effect of "sending an initial command to make the box listen for the actual command". Hence why I was asking if the arduino is maybe unable to capture the whole transmission from the remote. I used the components described in your site.

**Reply by bengtmartensson** [reply]
First of all; try sending the signals several times: Put Sending hw -> Count to 3 (for example).

Your captures look quite fine: All consists of a repeat and a ending, all of the same length. Signal "15" may not be correct though.

**Comment by edevil** [comment]
I tried several count values, even 100, and the box just ignored me... 🤷

**Reply by bengtmartensson** [reply]
That's bad. Actually, there are reasons to believe that the box is bluetooth/RF/CEC only: https://www.oneforall.com/support/faq/can-i-control-my-vodafone-giga-box-4k-one-all-remote-control https://forum.vodafone.de/t5/GigaTV/Vodafon-Giga-TV-4K-Box-mit-LG-MAGIC-REMOTE-UNIVERSAL/td-p/1931918 (Just two random Google hits, not a systematic search.)

Possibly a Flirc in the USB port would be a solution?

Another place to ask would be [remotecentral](http://www.remotecentral.com/cgi-bin/mboard/forums.cgi)

**Reply by edevil** [reply]
I had thought about this, but my universal remote is a Logitech Harmony 650 https://www.universalremotereviews.com/harmony-650-specifications/ AFAICT, it only supports IR. The remote I'm capturing the commands from is the Harmony, not the original box remote.

As for Flirc in the USB port, I'll investigate. Thanks!

**Comment by bengtmartensson** [comment]
Anything new in this?

**Comment by edevil** [comment]
I was never able to make it work with that vodafone box...
